opensubtitles/authentication.go

```diff
@@ -0,0 +1,40 @@
+package opensubtitles
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+// AuthenticationService provides access to the login related functions
+// in the OpenSubtitles API.
+//
+// OpenSubtitles API docs: https://www.opensubtitles.com/docs/api/html/index.htm#authentication
+type AuthenticationService service
+
+type LoggedIn struct {
+	User   UserData `json:"user"`
+	Token  string   `json:"token"`
+	Status int      `json:"status"`
+}
+
+//Login endpoint provides an authentication token to the rest of the API
+//
+//The response will return an token which should be included in all API requests to the server in a header that looks like the following:
+//
+//Authorization: your-auth-token
+func (s *AuthenticationService) Login(ctx context.Context, opt *Credentials) (loggedIn *LoggedIn, resp *http.Response, err error) {
+	u := "/api/v1/login"
+	payload := fmt.Sprintf("{\n  \"username\": \"%s\",\n  \"password\": \"%s\"\n}", opt.Username, opt.Password)
+	req, err := s.client.NewRequest("POST", u, payload)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	resp, err = s.client.Do(ctx, req, &loggedIn)
+	if err != nil {
+		return nil, resp, err
+	}
+	return
```

Comment:
Codacy found an issue: [extra empty line at the end of a block](https://app.codacy.com/manual/TheForgotten69/go-opensubtitles/pullRequest?prid=6093402)